homeserver_dendrite.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2022 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+)
+
+// Dendrite specific stats
+type ReportStatsDendrite struct {
+	// We're using mostly Synapse defined fields
+	Common             ReportStatsSynapse
```

Comment:
Considering you're already not persisting the Python version for Dendrite instances, I think it would make sense to separate out the actual common stats into a `ReportStatsCommon` struct. Presumably you can just introduce a `Common` field into `ReportStatsSynapse` and do something similar to what you've done here?